tests/unit/test_braille/test_routing.py

```diff
@@ -147,3 +147,58 @@ def test_moveCaret_always_instantActivate(self):
 		self.assertGreaterEqual(self.cm.lastActivateTime, curTime)
 		caret = self.cm.makeTextInfo(textInfos.POSITION_CARET)
 		self.assertEquals(caret, review)
+
+
+class TestTextInfoRegionRouting(unittest.TestCase):
+	"""A test for TextInfoRegion.getTextInfoForBraillePos, which is used in braille cursor routing.
+	This test ensures that braille routes to the expected character when dealing with emoji
+	or other composites.
+	These glyphs are threated as one character by uniscribe, however they span multiple characters
+	on a braille display.
+	Note that due to the nature of this test, it relies on uniscribe to be available.
+	"""
+
+	def test_routeToEmoji(self):
+		testText = "⚠️test"
+		obj = BasicTextProvider(text=testText)
+		ti = obj.makeTextInfo(textInfos.POSITION_CARET)
+		ti.expand(textInfos.UNIT_CHARACTER)
+		self.assertEqual(ti.text, testText[:2])
+		ti.collapse(end=True)
+		ti.expand(textInfos.UNIT_CHARACTER)
+		self.assertEqual(ti.text, testText[2])
+		region = braille.TextInfoRegion(obj)
+		region.update()
+		index = 3  # Position of e
+		pos = region.rawToBraillePos[index]
+		region.routeTo(pos)
+		ti = obj.makeTextInfo(textInfos.POSITION_CARET)
+		ti.expand(textInfos.UNIT_CHARACTER)
+		self.assertEqual(ti.text, testText[index])
+
+	def test_routeToComposite(self):
+		testText = "רבְּר"
+		obj = BasicTextProvider(text=testText)
+		ti = obj.makeTextInfo(textInfos.POSITION_CARET)
+		ti.expand(textInfos.UNIT_CHARACTER)
+		self.assertEqual(ti.text, testText[0])
+		ti.collapse(end=True)
+		ti.expand(textInfos.UNIT_CHARACTER)
+		self.assertEqual(ti.text, testText[1:4])
+		ti.collapse(end=True)
+		ti.expand(textInfos.UNIT_CHARACTER)
+		self.assertEqual(ti.text, testText[4])
+		region = braille.TextInfoRegion(obj)
+		region.update()
+		index = 1  # Position of ב
+		pos = region.rawToBraillePos[index]
+		region.routeTo(pos)
+		ti = obj.makeTextInfo(textInfos.POSITION_CARET)
+		ti.expand(textInfos.UNIT_CHARACTER)
+		self.assertEqual(ti.text, testText[1:4])
+		index = 3  # Position of ּ
```

Comment:
position of?

Reply:
> position of?

```suggestion
		index = 3  # Position of  ּ (\u5bc)
```

Reply:
Oh - I realized that this symbol is visually identical to my whitespace indicator for spaces, so this read to me as two spaces. Good reason to review with NVDA more! Thanks for adding the clarification, it's hard to see this character visually

Reply:
It works both ways, good that you brought this up and that it is now clear to all parties, both sighted and sr users